### PR TITLE
Add 'sic' function to bypass version number normalization

### PR DESCRIPTION
**Pull request by jaraco**
## Summary of changes

As discussed in #308, this proposed change adds a simple API to bypass version number normalization, forcing the user to explicitly opt into this undesirable behavior. Now a user can specify:

```
from setuptools import setup, sic

setup(version=sic('1.0.0a'))
```

And version number normalization will be skipped.

### Pull Request Checklist
- [x] Changes have tests
- [X] News fragment added in changelog.d. See [documentation](http://setuptools.readthedocs.io/en/latest/developer-guide.html#making-a-pull-request) for details

### CODE REVIEW

File: setuptools/dist.py [hunk 1]
```diff
@@ -438,9 +439,22 @@ def __init__(self, attrs=None):
...
             setattr(self.metadata, option, value)
 
-        self.metadata.version = self._validate_version(self.metadata.version)
+        self.metadata.version = self._normalize_version(
+            self._validate_version(self.metadata.version))
         self._finalize_requires()
 
+    @staticmethod
+    def _normalize_version(version):
+        if isinstance(version, setuptools.sic) or version is None:
```
Comment (pombredanne) [inline setuptools/dist.py]: What happens if `version=''` as opposed to `None` or is not a string?

Comment (jaraco) [reply setuptools/dist.py]: I would expect in those cases for the behavior to be unchanged with this PR.

**Comment by jaraco** [comment]
> Because I like to nitpick a bit, `verbatim` may be a more common Latin word than `sic` though? ... but that's purely cosmetic. An alternative could be `exactly()` or `as_is()` ?

I do strongly prefer a single-word names for functions where possible. I think `verbatim` or `exactly` would be suitable, but as I thought about it more, I like `sic` better because (a) the fact that it's a little esoteric in source code is jarring, and that's good for this case because it is a discouraged use and (b) the word `sic` inherently carries a connotation of "abnormal", so more than just "I want it this way,", it forces the author to acknowledge "I want it this way and that's not normal."

Good suggestions though, and compelling enough to warrant serious consideration and reflection.

**Comment by pombredanne** [comment]
@jaraco thank you++ :bowing_man:

**Pull request merged**